modules/shortcodes/polldaddy.php

```diff
@@ -181,26 +258,49 @@ function polldaddy_shortcode( $atts ) {
 							$margins = 'margin: 0px 0px 0px 10px';
 					}
 
+					// Force the normal style embed on single posts/pages otherwise it's not rendered on infinite scroll themed blogs ('infinite_scroll_render' isn't fired)
+					if ( is_singular() )
+						$inline = true;
+
 					if ( $cb === false && !$inline ) {
 						if ( self::$scripts === false )
 							self::$scripts = array();
 
 						$data = array( 'url' => $poll_js );
 
-						self::$scripts['poll'][] = $data;
+						self::$scripts['poll'][intval( $poll )] = $data;
```

Comment:
missing spaces in the $poll bracket.  Also should we first define this `$poll = intval( $poll )`?